File: app/src/main/java/com/kazakago/cleanarchitecture/domain/usecase/WeatherUseCase.java
package com.kazakago.cleanarchitecture.domain.usecase;

import com.kazakago.cleanarchitecture.domain.model.weather.WeatherModel;

import io.reactivex.Single;

/**
 * Weather UseCase
 *
 * @author Kensuke
 */
public interface WeatherUseCase {

    Single<WeatherModel> fetch(String cityId);

    WeatherModel find(String cityId);

}

File: app/src/main/java/com/kazakago/cleanarchitecture/domain/usecase/WeatherUseCase.kt
package com.kazakago.cleanarchitecture.domain.usecase

import com.kazakago.cleanarchitecture.domain.model.weather.WeatherModel

import io.reactivex.Single

/**
 * Weather UseCase

 * @author Kensuke
 */
interface WeatherUseCase {

    fun fetch(cityId: String): Single<WeatherModel>

    fun find(cityId: String): WeatherModel?

}